java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -256,11 +267,60 @@ private void visit(Tree tree, Tree terminator) {
           programState = put(programState, ((IdentifierTree) assignmentExpressionTree.variable()).symbol(), value);
         }
         break;
+      case METHOD_INVOCATION:
+        setSymbolicValueOnNonLocalVariables((MethodInvocationTree) tree);
+        break;
       default:
     }
     checkerDispatcher.executeCheckPreStatement(tree);
   }
 
+  private void setSymbolicValueOnNonLocalVariables(MethodInvocationTree tree) {
+    if (isLocalMethodInvocation(tree)) {
+      resetNullValuesOnNonLocalVariables(tree);
+    }
+  }
+
+  private static boolean isLocalMethodInvocation(MethodInvocationTree tree) {
+    ExpressionTree methodSelect = tree.methodSelect();
+    if (methodSelect.is(Tree.Kind.IDENTIFIER)) {
+      return true;
+    } else if (methodSelect.is(Tree.Kind.MEMBER_SELECT)) {
+      MemberSelectExpressionTree memberSelectExpression = (MemberSelectExpressionTree) methodSelect;
+      ExpressionTree target = memberSelectExpression.expression();
+      if (target.is(Tree.Kind.IDENTIFIER)) {
+        IdentifierTree identifier = (IdentifierTree) target;
+        return THIS_SUPER.contains(identifier.name());
+      }
+    }
+    return false;
+  }
+
+    private void resetNullValuesOnNonLocalVariables(MethodInvocationTree tree) {
+        boolean changed = false;
+        Map<Symbol, SymbolicValue> values = Maps.newHashMap(programState.values);
+        for (Entry<Symbol, SymbolicValue> entry : values.entrySet()) {
+            if (constraintManager.isNull(programState, entry.getValue())) {
+                Symbol symbol = entry.getKey();
+                if (isInstanceVariable(symbol)) {
+                    VariableTree variable = ((Symbol.VariableSymbol) symbol).declaration();
+                    if (variable != null) {
+                        changed = true;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 13. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)